src/rewriter.coffee

```diff
@@ -159,31 +159,36 @@ exports.Rewriter = class Rewriter
       # class declaration or if-conditionals)
       inImplicitControl = -> inImplicit() and stackTop()?[0] is 'CONTROL'
 
-      startImplicitCall = (j) ->
-        idx = j ? i
+      startImplicitCall = (idx) ->
         stack.push ['(', idx, ours: yes]
         tokens.splice idx, 0, generate 'CALL_START', '('
-        i += 1 if not j?
 
       endImplicitCall = ->
         stack.pop()
         tokens.splice i, 0, generate 'CALL_END', ')', ['', 'end of input', token[2]]
         i += 1
 
-      startImplicitObject = (j, startsLine = yes) ->
-        idx = j ? i
+      startImplicitObject = (idx, startsLine = yes) ->
         stack.push ['{', idx, sameLine: yes, startsLine: startsLine, ours: yes]
         val = new String '{'
         val.generated = yes
         tokens.splice idx, 0, generate '{', val, token
-        i += 1 if not j?
 
       endImplicitObject = (j) ->
         j = j ? i
         stack.pop()
         tokens.splice j, 0, generate '}', '}', token
         i += 1
 
+      implicitObjectContinues = (j) =>
+        nextTerminatorIdx = null
+        @detectEnd j,
```

Comment:
@vendethiel nope only gets called when it sees a `POST_IF` on the first line of an implicit object literal